olm/operator-bundle.md

```diff
@@ -0,0 +1,281 @@
+
+By standardizing on a container format for this data, we get many other features for free, such as: identity, distribution, replication, deduplication, signing, and ingress.
+
+### Goals
+* Define a convention for storing operator manifests and metadata with container image.
+* Build and push metadata using standard container tooling (e.g.docker cli)
+* No union filesystem should be required to consume metadata
+* Have a simple mechanism to apply a bundle to a kubernetes cluster
+
+### Non-Goals
+* Require OCI registries that support any non-standard media types
+* Build on-cluster tooling to interact with bundles
+
+## Proposal
+We delineate the operator metadata from the operator manifests. The operator manifests refers to a set of kubernetes manifest(s) the defines the deployment and RBAC model of the operator. The operator metadata on the other hand are, but not limited to:
+* Information that identifies the operator, it's name, version etc.
+* Additional information that drives the UI: 
+    * Icon
+    * Example CR(s)
+* Channel(s)
+* API(s) provided and required. 
+* Related images.
+
+
+This enhancement proposal focuses on the following:
+* A standard way to store and transmit manifests and metadata associated with an operator.
+* An operator author can specify supporting metadata in a standard and structured manner.
+* A single unique identifier that points to a particular version of an operator bundle (both metadata and manifests).
+
+The following user stories discuss the User Experience.
+
+---
+
+### User Stories
+
+#### Build, Push, Pull Operator Bundle
+As an operator author, I would like to associate operator manifests and metadata with the container image of my operator.
+
+The focus of this user story is to define a standard to store, transmit, inspect and retrieve operator manifests and metadata. The exact format of the metadata is outside of the scope of this story.
+
+*Constraints*:
+* An operator bundle (including both manifests and metadata) should be identifiable using a single versioned identifier. 
+* For an operator The metadata can be downloaded independently of the manifest.
+
+### Implementation Details/Notes/Constraints
+* The initial implementation target will be Docker v2-2 `manifests`, `manifest-lists`, and docker client support, for maximum compatiblity with existing tooling.
+* We want the entire operator bundle to be identifiable and retrievable using the same identifier/URL.
+
+#### Docker
+
+##### Build, Push, Pull Operator Bundle Image
+We use the following labels to annotate the operator bundle image.
+* The label `operators.operatorframework.io.bundle.resources` represents the bundle type:
+    * The value `manifests` implies that this bundle contains operator manifests only.
+    * The value `metadata` implies that this bundle has operator metadata only.
+    * The value `manifests+metadata` implies that this bundle contains both operator metadata and manifests.
```

Comment:
If there's not a direct need for this now, I suggest leaving it out of the initial implementation. I'm sure you will version the format you're creating, and if this turns out to be a valuable distinction, you can add it in a later version.

Otherwise, this adds a fair amount of complexity that may not be worthwhile.

Reply:
I agree with Michael, this is not something that we need to be focusing on.